src/compute/src/memory_management/memory_manager.rs

```diff
@@ -68,9 +70,15 @@ impl GlobalMemoryManager {
         watermark_epoch.store(epoch, Ordering::Relaxed);
     }
 
+    /// Jemalloc is not supported on non-Linux OSs, because tikv-jemalloc is not available.
+    /// See the comments for the macro enable_jemalloc_on_linux!();
+    #[cfg(not(target_os = "linux"))]
+    pub async fn run(self: Arc<Self>, _: Arc<BatchManager>, _: Arc<LocalStreamManager>) {}
```

Comment:
This is dangerous. A RisingWave without memory manager will continue to consume more memory until OOM 🥵

Reply:
Oh! That's not good. I guess we should warn users about this for non-Linux? I do not expect production-level usages of RisingWave on Windows/macOS, so I think this behavior _might_ be fine (idk, it would be nice if we can implement it with the default allocator)?

Reply:
It's possible to get rid of jemalloc after the competition of our new memory manager (#7180), because we will only rely on the memory usage collected by ourself (with `TaskLocalAlloc`)